File: ios/VisioMobile/i18n/Strings.swift
import Foundation

enum Strings {
    static let supportedLangs = ["en", "fr", "de", "es", "it", "nl"]
    private static var translations: [String: [String: String]] = [:]

    static func initialize() {
        for lang in supportedLangs {
            guard let url = Bundle.main.url(forResource: lang, withExtension: "json", subdirectory: "i18n"),
                  let data = try? Data(contentsOf: url),
                  let dict = try? JSONSerialization.jsonObject(with: data) as? [String: String]
            else { continue }
            translations[lang] = dict
        }
    }

    static func t(_ key: String, lang: String) -> String {
        return translations[lang]?[key] ?? translations["en"]?[key] ?? key
    }

    static func detectSystemLang() -> String {
        let sysLang = Locale.current.language.languageCode?.identifier ?? "en"
        return supportedLangs.contains(sysLang) ? sysLang : "en"
    }
}

File: android/app/src/main/kotlin/io/visio/mobile/ui/i18n/Strings.kt
package io.visio.mobile.ui.i18n

import android.content.Context
import org.json.JSONObject

object Strings {
    private var translations: MutableMap<String, Map<String, String>> = mutableMapOf()
    val supportedLangs = listOf("en", "fr", "de", "es", "it", "nl")

    fun init(context: Context) {
        for (lang in supportedLangs) {
            try {
                val json = context.assets.open("i18n/$lang.json").bufferedReader().readText()
                val obj = JSONObject(json)
                val map = mutableMapOf<String, String>()
                obj.keys().forEach { key -> map[key] = obj.getString(key) }
                translations[lang] = map
            } catch (_: Exception) {}
        }
    }

    fun t(key: String, lang: String): String {
        return translations[lang]?.get(key) ?: translations["en"]?.get(key) ?: key
    }

    fun detectSystemLang(): String {
        val sysLang = java.util.Locale.getDefault().language
        return if (sysLang in supportedLangs) sysLang else "en"
    }
}
